packages/quick_actions/quick_actions_ios/ios/Classes/QuickActionsPlugin.swift

```diff
@@ -0,0 +1,90 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import Flutter
+
+public final class QuickActionsPlugin: NSObject, FlutterPlugin {
+
+  public static func register(with registrar: FlutterPluginRegistrar) {
+    let channel = FlutterMethodChannel(
+      name: "plugins.flutter.io/quick_actions_ios",
+      binaryMessenger: registrar.messenger())
+    let instance = QuickActionsPlugin(channel: channel)
+    registrar.addMethodCallDelegate(instance, channel: channel)
+    registrar.addApplicationDelegate(instance)
+  }
+
+  private let channel: FlutterMethodChannel
+  private let shortcutStateManager: FLTShortcutStateManager
+  /// The type of the shortcut item selected when launching the app.
+  private var launchingShortcutType: String? = nil
+
+  // TODO: (hellohuanlin) remove `@objc` attribute and make it non-public after migrating tests to Swift.
+  @objc
+  public init(
+    channel: FlutterMethodChannel,
+    shortcutStateManager: FLTShortcutStateManager = FLTShortcutStateManager()
+  ) {
+    self.channel = channel
+    self.shortcutStateManager = shortcutStateManager
+  }
+
+  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
+    switch call.method {
+    case "setShortcutItems":
+      // `arguments` must be an array of dictionaries
+      let items = call.arguments as! [[String: Any]]
+      shortcutStateManager.setShortcutItems(items)
+      result(nil)
+    case "clearShortcutItems":
+      shortcutStateManager.setShortcutItems([])
+      result(nil)
+    case "getLaunchAction":
+      result(nil)
+    case _:
+      result(FlutterMethodNotImplemented)
+    }
+  }
+
+  public func application(
+    _ application: UIApplication, performActionFor shortcutItem: UIApplicationShortcutItem,
+    completionHandler: @escaping (Bool) -> Void
+  ) -> Bool {
+    handleShortcut(shortcutItem.type)
+    return true
+  }
+
+  public func application(
+    _ application: UIApplication,
+    didFinishLaunchingWithOptions launchOptions: [AnyHashable: Any] = [:]
+  ) -> Bool {
+    if let shortcutItem = launchOptions[UIApplication.LaunchOptionsKey.shortcutItem]
+      as? UIApplicationShortcutItem
+    {
+      // Keep hold of the shortcut type and handle it in the
+      // `applicationDidBecomeActure:` method once the Dart MethodChannel
```

Comment:
This was presumably meant to say "Active", not "Acture"; we shouldn't preserve the typo.